### PR TITLE
Add rough draft of per-file techMD-related ADR

### PR DESCRIPTION
**Pull request by mjgiarlo**

### CODE REVIEW

File: 0003-extract-technical-metadata-on-per-file-basis.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  * ...
+* Date(s): <!-- required -->
+  * proposed: 2019-10-29
+  * ...
+
+## Context and Problem Statement <!-- required -->
+
+Currently, we extract technical metadata per-object and run one extraction job serially per-file. This takes a problematically long time for objects with many files; blocks other objects from accessioning; and complicates restarts which must begin again and process the entire object.
+
+NOTE: Needs discussion: Fedora 3 does not support concurrent writes on the same datastream so we can either split out filesets as a first-class objects in the F3 data model or use temporary caching to generate a consolidated techMD datastream.
```
Comment (justinlittman) [inline 0003-extract-technical-metadata-on-per-file-basis.md]: Can we parallelize within a robot? That is, a single robot initiates, manages, and merges multiple per-file technical metadata processes for a single object.

Comment (mjgiarlo) [reply 0003-extract-technical-metadata-on-per-file-basis.md]: I'm not sure. That could be a promising implementation indeed. Thanks for reviewing and commenting!

Comment (jcoyne) [reply 0003-extract-technical-metadata-on-per-file-basis.md]: @justinlittman I don't think we can do a merge within a robot.  I think we need to track (in a database or elsewhere) all the parts, and then do the merge job when all parts are complete. I'm open to other suggestions.

Comment (justinlittman) [reply 0003-extract-technical-metadata-on-per-file-basis.md]: Couldn't we just use https://github.com/grosser/parallel or similar?

Comment (jcoyne) [reply 0003-extract-technical-metadata-on-per-file-basis.md]: @justinlittman that won't spread the load across multiple machines.

Comment (justinlittman) [reply 0003-extract-technical-metadata-on-per-file-basis.md]: Oh, that is not at all clear in this ADR that want to spread the load across multiple machines. Is that overkill for this problem? Have we evaluated the simpler approach of parallelizing on a single machine?

Comment (justinlittman) [reply 0003-extract-technical-metadata-on-per-file-basis.md]: Side note: Surfacing issues like this is exactly why we should write ADRs, so yeah!

Comment (jcoyne) [reply 0003-extract-technical-metadata-on-per-file-basis.md]: Another side note: In the cocina design phase we determined that individual files should be addressable (e.g. have an id and can be looked up and updated individually).  If we start storing tech metadata per file, we can  adopt this approach, which would help us evolve away from being coupled to Fedora 3.

**Comment by justinlittman** [comment]
So I ran a test of parallelizing tech md generation (using JHOVE) on a single machine. A test involved performing analysis of the JP2s for a single google book (181 files). Each test was run 5 times and the following are the averages:
```
1 threads averaged 114.21 secs
2 threads averaged 64.74 secs
3 threads averaged 49.49 secs
4 threads averaged 43.64 secs
5 threads averaged 40.33 secs
6 threads averaged 38.71 secs
7 threads averaged 37.64 secs
8 threads averaged 38.70 secs
9 threads averaged 38.94 secs
10 threads averaged 38.80 secs
1 processes averaged 111.78 secs
2 processes averaged 59.74 secs
3 processes averaged 45.25 secs
4 processes averaged 39.90 secs
5 processes averaged 37.53 secs
6 processes averaged 36.56 secs
7 processes averaged 36.49 secs
8 processes averaged 36.29 secs
9 processes averaged 35.94 secs
10 processes averaged 35.84 secs
```

These results suggest significant improvements from parallelizing on a single machine.

**Pull request merged**